include/dlaf/util_math.h

```diff
@@ -10,19 +10,34 @@
 
 #pragma once
 
+#include <type_traits>
+
 /// @file
 
 namespace dlaf {
 namespace util {
 
-/// @brief Returns @c ceiling(@p num / @p den) for integer types.
+/// @brief Returns ceiling(num/den) for integer types.
+///
+/// It accepts both signed and unsigned integer types, but as precondition states both values must
+/// not be negative (additionally, @p den must not be zero either)
+///
+/// @tparam IntType must be a signed or unsigned integer type.
+/// @param num
+/// @param den
+/// @return ceiled division of type @a IntType
 ///
-/// @tparam IntType has to be an integer type
-/// @pre @a num >= 0 and @a den >= 0
-template <class IntType>
-constexpr IntType ceilDiv(IntType num, IntType den) {
+/// @pre @a num >= 0 and @a den > 0.
+/// @pre @a num + @a den - 1 must not overflow @a IntType max value
+template <typename IntType>
+constexpr auto ceilDiv(const IntType num, const IntType den)
+#ifdef DLAF_DOXYGEN
+    ;
```

Comment:
I think that for readability `#if` macros should not contain just a part of a line of code.

For doxygen I suggest to use
```
#ifdef DLAF_DOXYGEN
  full function definition (including templates) to be shown in doxygen
#else
  function definition or declaration to be compiled.
#endif
```
Also modify the doxygen section of the coding style to include this hint.